config/http_config.go

```diff
@@ -172,6 +176,9 @@ func (c *HTTPClientConfig) Validate() error {
 // UnmarshalYAML implements the yaml.Unmarshaler interface
 func (c *HTTPClientConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	type plain HTTPClientConfig
+	*c = HTTPClientConfig{
```

Comment:
This overrides anything that might be pre-populated in `*c`. Not sure if we ever do that, but wouldn't it be safer (and easier) to do `c.FollowRedirects = true`?

Reply:
No because C is nil at this point.

We do this in Prometheus:

https://github.com/prometheus/prometheus/blob/537c0aff49f14f9eed067c6a54a8edac862273b5/config/config.go#L191

Reply:
I have switched to the exact same pattern as prometheus (expose default struct)

Reply:
Just FTR: If `c` is `nil`, then L179, as above, panics.

Reply:
mmmmh yes indeed. but still, this is the pattern we use everywhere, let's stick to it?